pkg/Library/Shader/ShaderFragment.js

```diff
@@ -47,11 +47,17 @@
       }
     });
   },
-  render({}, {output}) {
+  render({shader}, {output}) {
     return {
-      output
+      output,
+      code: shader || ''
     };
   },
+  onCodeChanged({eventlet: {key, value}}) {
+    if (key === 'code') {
```

Comment:
there's no other source for this event, I don't think you need this test

Reply:
done